homeassistant/components/vlc_telnet/media_player.py

```diff
@@ -249,6 +249,10 @@ def set_volume_level(self, volume):
         self._vlc.set_volume(volume * MAX_VOLUME)
         self._volume = volume
 
+        if self._muted and self._volume > 0:
+            # This can happen if we were muted and then see a volume_up.
+            self._muted = False
```

Comment:
Is there no feedback on mute state that we can get from vlc on update?

Reply:
Yeah, good question. I've looked at the output of `VLCTelnet.status()` and `VLCTelnet.info()`, but didn't find anything relevant. AFAICT, the VLC telnet protocol doesn't ever tell us if VLC is actually muted so this seems to be a deeper problem.

Seems we'd want to use https://pypi.org/project/python-vlc/ for this type of control.

Reply:
Ok, that library is using the c-bindings so is out of scope for this integration.

Instead of using a separate variable for muted I suggest we check if `self._volume` equals 0. The volume attribute is updated by the update method.

Reply:
Hm, not sure I understand. Would that check mean that if the user sets the volume to 0 in VLC, Home Assistant would detect VLC as muted? If so, what would the behavior be if the user clicks unmute in Home Assistant?

Incidentally, here's another possible direction: We remove `SUPPORT_VOLUME_MUTE` from `vlc_telnet` entirely (since VLC doesn't *actually* support this and we're essentially simulating mute...).

(and yes, switching to the CTypes lib would be a totally separate integration -- probably not worth it if the only benefit is adding mute support :))

Reply:
Good point. We probably wouldn't have added mute support today simulated like this, but removing it now is also a breaking change. We have some simulated features in the base media player class so we're a bit more flexible it seems for the media player. I'm ok with this fix.

Reply:
The vlc integration uses the c-bindings library.

https://www.home-assistant.io/integrations/vlc/

Reply:
Thanks Martin!

> The vlc integration uses the c-bindings library.

Indeed! I'll have to revisit using that integration in my setup -- I forget why I settled on the telnet one of all things :)

Reply:
You need to have the vlc application installed on the same host as Home Assistant for that integration. I think that may be a major limitation for some users.